replica/table.cc

```diff
@@ -577,16 +577,16 @@ class permit_monitor : public sstables::write_monitor {
 // Handles all tasks related to sstable writing: permit management, compaction backlog updates, etc
 class database_sstable_write_monitor : public permit_monitor, public backlog_write_progress_manager {
     sstables::shared_sstable _sst;
-    sstables::compaction_strategy& _compaction_strategy;
+    compaction_group& _compaction_group;
     const sstables::writer_offset_tracker* _tracker = nullptr;
     uint64_t _progress_seen = 0;
     api::timestamp_type _maximum_timestamp;
 public:
     database_sstable_write_monitor(lw_shared_ptr<sstable_write_permit> permit, sstables::shared_sstable sst,
-        sstables::compaction_strategy& strategy, api::timestamp_type max_timestamp)
+        compaction_group& cg, api::timestamp_type max_timestamp)
             : permit_monitor(std::move(permit))
             , _sst(std::move(sst))
-            , _compaction_strategy(strategy)
+            , _compaction_group(cg)
             , _maximum_timestamp(max_timestamp)
```

Comment:
Maybe we should move database_sstable_write_monitor to compaction_manager. After all, the only reason database is interested in it is to set compaction shares.

Reply:
It's only used in the context of memtable flush. compaction.cc has compaction_write_monitor, which is similar to this monitor but only used by compaction.

It stores compaction_group, which lives in replica namespace, because compaction_group stores many replica entities like memtable. So I think it's a layer violation to move it to compaction/

Reply:
Please disconsider the last paragraph. We could switch to table_state if we really want to move database_sstable_write_monitor to compaction/.

Reply:
Well, memtable flush isn't really in the scope of compaction_manager. So it's my comment that was wrong, not yours.